src/Sylius/Bundle/CartBundle/Controller/CartItemController.php

```diff
@@ -137,4 +135,47 @@ public function removeAction(Request $request)
 
         return $this->redirectToCartSummary($configuration);
     }
+
+    /**
+     * @param string $flashMessage
+     * @param MetadataInterface $metadata
+     *
+     * @return string
+     */
+    private function translateMessage($flashMessage, MetadataInterface $metadata)
+    {
+        return $this->get('translator')->trans($flashMessage, ['%resource%' => ucfirst($metadata->getHumanizedName())], 'flashes');
+    }
+
+    /**
+     * @param CartInterface $cart
+     */
+    private function recalculateCart(CartInterface $cart)
+    {
+        $orderRecalculator = $this->get('sylius.order_processing.order_recalculator');
+        $orderRecalculator->recalculate($cart);
+
+        $cartManager = $this->get('sylius.manager.cart');
+        $cartManager->persist($cart);
+        $cartManager->flush();
+    }
+
+    /**
+     * @param CartInterface $cart
+     * @param CartItemInterface $item
+     */
+    private function resolveCartItem(CartInterface $cart, CartItemInterface $item)
+    {
+        foreach ($cart->getItems() as $existingItem) {
+            if ($item->equals($existingItem)) {
+                $itemQuantityModifier = $this->get('sylius.order_item_quantity_modifier');
+                $itemQuantityModifier->modify($existingItem, $existingItem->getQuantity() + $item->getQuantity());
+
+                return;
+            }
+        }
+
+        $cart->addItem($item);
+        $this->repository->add($item);
```

Comment:
This seems wrong, we should add the item to cart and persist + flush the cart itself, not item.